### PR TITLE
k8s/controller: always map "default" Consul namespace to empty string

### PR DESCRIPTION
**Pull request by mikemorris**
### Changes proposed in this PR:

Always map the `"default"` Consul namespace to the empty string in the memory backend store, for compatibility with Consul OSS and the logic to [parse the namespace of a Gateway from the SPIFFE URI](https://github.com/hashicorp/consul-api-gateway/blob/v0.5.1/internal/envoy/middleware.go#L112-L114), used when [reading a Gateway from a store backend](https://github.com/hashicorp/consul-api-gateway/blob/v0.5.1/internal/envoy/middleware.go#L79-L85).

This Consul namespace mapper function is already called when [reconciling a Gateway to a backend store](https://github.com/hashicorp/consul-api-gateway/blob/v0.5.1/internal/k8s/reconciler/manager.go#L180-L192), but the corresponding lookup fails with Consul Enterprise when `--consul-destination-namespace` is set to `"default"` because of the special case handling in the `parseURI` helper.

No migration handling should be needed because this is all within the ephemeral memory store for now, but this does need an e2e test for a service in the `"default"` namespace on Consul Enterprise.

As a temporary workaround, this edge case can be avoided by configuring `consulNamespaces.mirroringK8S: true` or specfiying `""` or _any_ value other than `"default"` for `consulNamespaces.consulDestinationNamespace` in the Helm chart.

### Checklist:
- [x] Tests added
- [x] CHANGELOG entry added 
  > Run `make changelog-entry` for guidance in authoring a changelog entry, and
  > commit the resulting file, which should have a name matching your PR number.
  > Entries should use imperative present tense (e.g. Add support for...)

### CODE REVIEW

File: internal/commands/server/k8s_e2e_test.go [hunk 1]
```diff
@@ -116,6 +116,69 @@ func TestGatewayWithClassConfigChange(t *testing.T) {
...
+			checkGatewayConfigAnnotation(ctx, t, resources, thirdGatewayName, namespace, gcc)
+
+			// Cleanup
+			assert.NoError(t, resources.Delete(ctx, firstGateway))
+			assert.NoError(t, resources.Delete(ctx, secondGateway))
+			assert.NoError(t, resources.Delete(ctx, thirdGateway))
+
+			// Re-enable namespace mirroring
+			ctx, err = e2e.SetNamespaceMirroring(true)(ctx, nil)
+			require.NoError(t, err)
```
Comment (mikemorris) [inline internal/commands/server/k8s_e2e_test.go:174]: I was surprised this would be necessary, as I expected SetUpStack to create a new context for each test, but it appears to be reused - I'm not sure if that's intentional/common, or if that should be changed and this cleanup could be removed?

Comment (sarahalsmiller) [reply internal/commands/server/k8s_e2e_test.go:174]: I've run into this issue before when touching the e2e tests as well. I'm not sure what the right solution is because I definitely think spinning up a new kind cluster for each test would make these tests take way longer, but the current set up make it difficult to do any kind of namespace manipulation.

**Review by sarahalsmiller — APPROVED** [review]
Thanks for hopping on this

**Review by andrewstucki — APPROVED** [review]
The patch itself looks fine to me
